src/tuple.cr

```diff
@@ -545,20 +545,16 @@ struct Tuple
   # {1, 2, 3, 4, 5}.to_a # => [1, 2, 3, 4, 5]
   # ```
   def to_a : Array(Union(*T))
-    {% if compare_versions(Crystal::VERSION, "1.1.0") < 0 %}
-      to_a(&.itself.as(Union(*T)))
-    {% else %}
-      to_a(&.itself)
-    {% end %}
```

Comment:
With this change, the explicit cast is always necessary, even after `1.1.0` because the compiler couldn't figure out the output type of `&.itself`.